### ISSUE TITLE
hasUnlimitedStock set to Null when "Don't import"

### ISSUE THREAD

**Issue by nitech** [craftcms/feed-me#636]
### Description
When I run a feed where the `Product Variant Fields'` `Unlimited Stock` has the default value `Don't import`, the Variant record in `commerce_variants.hasUnlimitedStock` is set to `Null`.

![image](https://user-images.githubusercontent.com/1279436/70124159-6bdec380-1674-11ea-9c29-64ae992c5e03.png)

This causes weirdness when you run queries like this: 

```twig
{% set children = 
    craft.products
        .productImage(":notempty:")
        .hasVariant({ hasStock: true })
        .limit(50) %}
```

```sql
SELECT `commerce_variants`.`productId`
FROM 
	(SELECT `elements`.`id` AS `elementsId`, `elements_sites`.`id` AS `elementsSitesId`, `content`.`id` AS `contentId`
	FROM `elements` `elements`
	INNER JOIN `commerce_variants` `commerce_variants` ON `commerce_variants`.`id` = `elements`.`id`
	LEFT JOIN `commerce_products` `commerce_products` ON `commerce_variants`.`productId` = `commerce_products`.`id`
	INNER JOIN `elements_sites` `elements_sites` ON `elements_sites`.`elementId` = `elements`.`id`
	INNER JOIN `content` `content` ON (`content`.`elementId` = `elements`.`id`) AND (`content`.`siteId` = `elements_sites`.`siteId`)
	WHERE 
		(
			((commerce_variants.stock > 0 AND commerce_variants.hasUnlimitedStock != 1)) 
			OR (commerce_variants.hasUnlimitedStock = 1)
		) 
		AND (`elements_sites`.`siteId`='1') 
		AND (`elements`.`archived`=FALSE) 
		AND (`elements`.`enabled`=TRUE) 
		AND (`elements`.`dateDeleted` IS NULL) 
		AND (`elements_sites`.`enabled`=TRUE) 
		AND (`elements`.`draftId` IS NULL) 
		AND (`elements`.`revisionId` IS NULL)
	ORDER BY `commerce_variants`.`sortOrder`) `subquery`
INNER JOIN `commerce_variants` `commerce_variants` ON `commerce_variants`.`id` = `subquery`.`elementsId`
INNER JOIN `elements` `elements` ON `elements`.`id` = `subquery`.`elementsId`
INNER JOIN `elements_sites` `elements_sites` ON `elements_sites`.`id` = `subquery`.`elementsSitesId`
INNER JOIN `content` `content` ON `content`.`id` = `subquery`.`contentId`
ORDER BY `commerce_variants`.`sortOrder`
```

Because of this sentence: 

```sql
((commerce_variants.stock > 0 AND commerce_variants.hasUnlimitedStock != 1)) 
```

If you modify the query by adding `ifnull(commerce_variants.hasUnlimitedStock,0)` to VariantQuery.php:470, you get around the problem:

```php
if (null !== $this->hasStock && (bool)$this->hasStock === true) {
    $hasStockCondition = ['or', '(commerce_variants.stock > 0 AND ifnull(commerce_variants.hasUnlimitedStock,0) != 1)', 'commerce_variants.hasUnlimitedStock = 1'];
    $this->subQuery->andWhere($hasStockCondition);
}
```
... but I reckon the fix should really be in the `Nullable` attribute of the table column `hasUnlimitedStock`, paired with an update on Feed Me's default handling of the default value `Don't import`

### Additional info

- Plugin version: 4.1.2
- Craft version: 3.3.16.3

**Comment by nitech** [comment]
I added a new feed and modified the default value to "Default value" and did not check the checkbox to the right -- but the field `hasUnlimitedStock` is still `Null`.

**Comment by joshangell** [comment]
I’m seeing this as well - latest versions of Craft, Commerce and Feed Me.

**Comment by itopiainteractive** [comment]
We seem to be hitting this issue when importing product data. No matter what we set in Feedme (e.g. Don't Import or Use Default) that hasUnlimitedStock value must be set in our feed, otherwise it throws a NULL error. This becomes especially critical when we don't have much control of the feed itself.

**Comment by i-just** [comment]
This was fixed in [Commerce 3.4.22 ](https://github.com/craftcms/commerce/blob/v3/CHANGELOG.md#3422---2023-05-31)
